src/test/org/firebirdsql/gds/impl/jni/TestServicesAPI.java

```diff
@@ -45,33 +45,31 @@
  * Initial tests for Services API. Currently run only against embedded server.
  * TODO: Make run against other types
  */
-public class TestServicesAPI {
-
-    @ClassRule
-    public static final GdsTypeRule testType = GdsTypeRule.supports(EmbeddedGDSFactoryPlugin.EMBEDDED_TYPE_NAME);
+public abstract class TestServicesAPI {
```

Comment:
I think the changes to this class (and the creation of its subclasses) can be avoided by making proper use of the GdsTypeRule and FBTestProperties.

Reply:
I'll try to do it.